Reflection-free-serializers serialization error

### Describe the bug

When using the function `quarkus.rest.jackson.optimization.enable-reflection-free-serializers=true`, an object handling error occurs.
I have an object:
```
data class OwnerDto(
    val id: Long?,
    val ownerId: String?,
    val ownerEmail: String?,
    val accessKey: String?,
    val accessStatus: Boolean?,
    val enableDate: LocalDateTime?,
)
```

Its serialization fails:
```
Caused by: java.lang.NoSuchMethodError: 'void com.fasterxml.jackson.core.JsonGenerator.writeBoolean(java.lang.Boolean)'
	at com.test.service.db.OwnerDto$quarkusjacksonserializer.serialize(Unknown Source)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serializeContents(IndexedListSerializer.java:119)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:79)
	at com.fasterxml.jackson.databind.ser.impl.IndexedListSerializer.serialize(IndexedListSerializer.java:18)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:502)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
	at com.fasterxml.jackson.databind.ObjectWriter$Prefetch.serialize(ObjectWriter.java:1574)
	at com.fasterxml.jackson.databind.ObjectWriter._writeValueAndClose(ObjectWriter.java:1275)
	at com.fasterxml.jackson.databind.ObjectWriter.writeValue(ObjectWriter.java:1098)
	at io.quarkus.resteasy.reactive.jackson.runtime.serialisers.BasicServerJacksonMessageBodyWriter.writeResponse(BasicServerJacksonMessageBodyWriter.java:63)
	at org.jboss.resteasy.reactive.server.core.ServerSerialisers.invokeWriter(ServerSerialisers.java:217)
```
Generated serializer code:
```
public class OwnerDto$quarkusjacksonserializer extends StdSerializer {
    public OwnerDto$quarkusjacksonserializer() {
        super(OwnerDto.class);
    }

    public void serialize(Object var1, JsonGenerator var2, SerializerProvider var3) throws IOException {
        OwnerDto var5 = (OwnerDto)var1;
        var2.writeStartObject();
        SerializedString var4 = quarkusjacksonserializer.accessKey;
        var2.writeFieldName((SerializableString)var4);
        String var6 = var5.getAccessKey();
        var2.writeString(var6);
        SerializedString var7 = quarkusjacksonserializer.enableDate;
        var2.writeFieldName((SerializableString)var7);
        LocalDateTime var8 = var5.getEnableDate();
        var2.writePOJO(var8);
        SerializedString var9 = quarkusjacksonserializer.id;
        var2.writeFieldName((SerializableString)var9);
        Long var10 = var5.getId();
        var2.writeNumber((Long)var10);
        SerializedString var11 = quarkusjacksonserializer.ownerEmail;
        var2.writeFieldName((SerializableString)var11);
        String var12 = var5.getOwnerEmail();
        var2.writeString(var12);
        SerializedString var13 = quarkusjacksonserializer.ownerId;
        var2.writeFieldName((SerializableString)var13);
        String var14 = var5.getOwnerId();
        var2.writeString(var14);
        SerializedString var15 = quarkusjacksonserializer.tAccessStatus;
        var2.writeFieldName((SerializableString)var15);
        Boolean var16 = var5.getAccessStatus();
        var2.writeBoolean(var16);
        var2.writeEndObject();
    }
}
```
However, if you make any changes to the code, a hot-reboot will be performed and this error will not occur.

### Expected behavior

Serialization works immediately without errors

### Actual behavior

Serialization error occurs in development mode before hot-reboot

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

java 22.0.2

### Quarkus version or git rev

3.14.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Gradle 8.10
------------------------------------------------------------

Build time:    2024-08-14 11:07:45 UTC
Revision:      fef2edbed8af1022cefaf44d4c0514c5f89d7b78

Kotlin:        1.9.24
Groovy:        3.0.22
Ant:           Apache Ant(TM) version 1.10.14 compiled on August 16 2023
Launcher JVM:  22.0.2 (Oracle Corporation 22.0.2+9-70)
Daemon JVM:    /Library/Java/JavaVirtualMachines/jdk-22.jdk/Contents/Home (no JDK specified, using current Java home)
OS:            Mac OS X 14.6.1 aarch64

### Additional information

Launch in IDEA 2024.2.1

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/main/java/io/quarkus/resteasy/reactive/jackson/deployment/processor/JacksonCodeGenerator.java
@@ -357,6 +357,7 @@ public abstract class JacksonCodeGenerator {
                 case "java.lang.Long" -> "long";
                 case "java.lang.Double" -> "double";
                 case "java.lang.Float" -> "float";
+                case "java.lang.Boolean" -> "boolean";
                 default -> fieldType.name().toString();
             };
         }

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/Dog.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/Dog.java
@@ -7,11 +7,22 @@ public class Dog extends AbstractNamedPet {
     @JsonProperty("age")
     private int publicAge;
 
+    @JsonProperty("vaccinated")
+    private Boolean publicVaccinated;
+
     public int getPublicAge() {
         return publicAge;
     }
 
     public void setPublicAge(int publicAge) {
         this.publicAge = publicAge;
+    }
+
+    public Boolean getPublicVaccinated() {
+        return publicVaccinated;
+    }
+
+    public void setPublicVaccinated(Boolean publicVaccinated) {
+        this.publicVaccinated = publicVaccinated;
     }
 }

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonResource.java
@@ -171,6 +171,7 @@ public class SimpleJsonResource extends SuperClass<Person> {
         dog.setPrivateName("Jack");
         dog.setPublicName("Leo");
         dog.setVeterinarian(createVeterinarian());
+        dog.setPublicVaccinated(true);
         return dog;
     }
 

--- a/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonTest.java
+++ b/extensions/resteasy-reactive/rest-jackson/deployment/src/test/java/io/quarkus/resteasy/reactive/jackson/deployment/test/SimpleJsonTest.java
@@ -661,7 +661,7 @@ public class SimpleJsonTest {
     public void testEcho() {
         RestAssured
                 .with()
-                .body("{\"publicName\":\"Leo\",\"veterinarian\":{\"name\":\"Dolittle\"},\"age\":5}")
+                .body("{\"publicName\":\"Leo\",\"veterinarian\":{\"name\":\"Dolittle\"},\"age\":5,\"vaccinated\":true}")
                 .contentType("application/json; charset=utf-8")
                 .post("/simple/dog-echo")
                 .then()
@@ -670,6 +670,7 @@ public class SimpleJsonTest {
                 .body("publicName", Matchers.is("Leo"))
                 .body("privateName", Matchers.nullValue())
                 .body("age", Matchers.is(5))
+                .body("vaccinated", Matchers.is(true))
                 .body("veterinarian.name", Matchers.is("Dolittle"))
                 .body("veterinarian.title", Matchers.nullValue());
     }

